js/src/components/paid-ads/asset-group/asset-field.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useReducedMotion } from '@wordpress/compose';
+import {
+	useState,
+	useRef,
+	useImperativeHandle,
+	forwardRef,
+} from '@wordpress/element';
+import { chevronUp, chevronDown } from '@wordpress/icons';
+import { Pill } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import AppButton from '.~/components/app-button';
+import HelpPopover from '.~/components/help-popover';
+import './asset-field.scss';
+
+/**
+ * @typedef {Object} AssetFieldHandler
+ * @property {() => void} scrollIntoComponent Scroll to the nearest alignable position to show this component in the visible area.
+ */
+
+/**
+ * Renders an expandable wrapper for editing asset fields.
+ *
+ * @param {Object} props React props.
+ * @param {JSX.Element} props.heading Heading.
+ * @param {JSX.Element} props.subheading Subheading.
+ * @param {JSX.Element} props.help Help content to be shown after clicking on the help icon.
+ * @param {number} props.numOfIssues The number of issues used to label on UI. It only labels when the number is greater than 0.
+ * @param {boolean} [props.initialExpanded=false] Whether the UI to initial with expanded content.
+ * @param {boolean} [props.disabled=false] Whether display the UI in disabled style. It will collapse the content when disabled.
+ * @param {JSX.Element} [props.children] Content to be rendered.
+ * @param {import('react').MutableRefObject<AssetFieldHandler>} ref React ref to be attached to the handler of this component.
+ */
+function AssetField(
+	{
+		heading,
+		subheading,
+		help,
+		numOfIssues,
+		initialExpanded = false,
+		disabled = false,
+		children,
+	},
+	ref
+) {
+	const containerRef = useRef();
+	const [ isExpanded, setExpanded ] = useState( initialExpanded );
```

Comment:
💅  I personally would rename this as `[ isExpanded, setIsExpanded ]` 

or

`[ expanded, setExpanded ]`

Reply:
In this codebase, there are many of the same namings but they didn't seem to have caused confusion before. I would like to keep it as is if it still doesn't cause confusion.

Reply:
I'm not aware of those and this is not a blocker. But for me the standard React way of calling it [property, setProperty] is more convenient. And here ,especially, since there are several expanded related variables (initalExpanded, shouldExpanded and isExpanded).

Reply:
Thanks for the explanation. Renamed in https://github.com/woocommerce/google-listings-and-ads/pull/1853/commits/a33c984fc67205c07344923348205a3991e4120c.

To clarify, I'm not pretty sure if there is a standard React way to name the returned pair of `useState`. My understanding is that it's only a convention but not a standard, and WP and WC also have some cases in the same naming way. However, following the most popular convention on the naming of `useState` is still better.